streams/src/test/java/org/apache/kafka/test/ReadOnlySessionStoreStub.java

```diff
@@ -82,11 +118,15 @@ public boolean hasNext() {
                 public KeyValue<Windowed<K>, V> next() {
                     return it.next();
                 }
-
             }
         );
     }
 
+    @Override
+    public KeyValueIterator<Windowed<K>, V> backwardFetch(K from, K to) {
+        return null;
```

Comment:
I guess it probably doesn't matter since we presumably aren't using these backward methods of the ReadOnlySessionStoreStub, but it seems like it might result in some tricky NPEs to debug if ever someone does try to use it in a test. If you don't feel like implementing it I think it's fine to just throw UnsupportedOperationException and say that you'll have to implement this to use it.

Or just copy the code from the forward direction and flip it 🤷‍♀️  Same goes for all the methods in here that return null

Reply:
mixed choice here. hope is good enough